src/DAG/dag_parser.c

```diff
@@ -35,42 +187,39 @@ static int DAG_ParseLoadArgs(RedisModuleCtx *ctx, RedisModuleString **argv, int
                                    "ERR invalid or negative value found in number of keys to LOAD");
         return -1;
     }
+
     int number_loaded_keys = 0;
-    int separator_flag = 0;
-    size_t argpos = 2;
-    for (; (argpos <= argc - 1) && (number_loaded_keys < n_keys); argpos++) {
-        size_t arg_len;
+    size_t arg_len;
+
+    // Go over the given args and load the tensors from keyspace.
+    for (size_t argpos = 2; argpos < argc && number_loaded_keys < n_keys; argpos++) {
         const char *arg_string = RedisModule_StringPtrLen(argv[argpos], &arg_len);
-        if (!strcasecmp(arg_string, chaining_operator)) {
-            separator_flag = 1;
+        if (!strcasecmp(arg_string, chaining_operator))
             break;
-        } else {
-            RAI_Tensor *t;
-            RedisModuleKey *key;
-            const int status =
-                RAI_GetTensorFromKeyspace(ctx, argv[argpos], &key, &t, REDISMODULE_READ);
-            if (status == REDISMODULE_ERR) {
-                RedisModule_Log(ctx, "warning",
-                                "on DAGRUN's LOAD could not load tensor %s from keyspace",
-                                arg_string);
-                return -1;
-            }
-            char buf[16];
-            sprintf(buf, "%04d", 1);
-            RedisModuleString *dictKey = RedisModule_CreateStringFromString(NULL, argv[argpos]);
-            RedisModule_StringAppendBuffer(NULL, dictKey, buf, strlen(buf));
-
-            AI_dictAdd(*localContextDict, (void *)dictKey, (void *)RAI_TensorGetShallowCopy(t));
-            AI_dictAdd(*loadedContextDict, (void *)dictKey, (void *)1);
-            RedisModule_FreeString(NULL, dictKey);
-            number_loaded_keys++;
+        RAI_Tensor *t;
+        RedisModuleKey *key;
+        const int status = RAI_GetTensorFromKeyspace(ctx, argv[argpos], &key, &t, REDISMODULE_READ);
+        if (status == REDISMODULE_ERR) {
+            RedisModule_Log(ctx, "warning",
+                            "on DAGRUN's LOAD could not load tensor %s from keyspace", arg_string);
+            return -1;
         }
+
+        // Add the tensor under its "mangled" key name to the DAG local context dict.
+        char buf[16];
+        sprintf(buf, "%04d", 1);
```

Comment:
can we skip this phase?
every tensor is loaded with `0001` suffix and then it is removed and replaced as far as I understand the code